### ISSUE TITLE
pre-enhanced app distributions

### ISSUE THREAD

**Issue by dragonwocky** [notion-enhancer/notion-enhancer#321]
this has been created as a sister-issue to https://github.com/davidbailey00/notion-deb-builder/pull/18 so it won't be forgotten.

after the module rework in v0.11.0 is done, it would be possible to create and distribute pre-enhanced installers for notion. this could definitely be done with the deb builder, but i'm not sure yet if could be done for macOS/windows (or if it needs to be done?).

>  in theory it would just need: copying the app folder to make an `app.bak`, appending a single `require()` statement to every js file in the `app` folder, and adding in an `enhancer` folder at the same level as the `app` folder.

**Comment by dragonwocky** [comment]
@davidbailey00 i'm not super familiar with the build process: would it be possible to also do this for macOS/windows versions of notion?

**Comment by davidbailey00** [comment]
I haven't tried it yet but I believe it would be possible yes!

**Comment by davidbailey00** [comment]
> appending a single `require()` statement to every js file in the app folder, and adding in an enhancer folder at the same level as the app folder.

it may be easier to put the enhancer folder inside `node_modules`, that way it can be imported using just `require('notion-enhancer/pkg/loader.js`)` without need to specify a relative path.

also, is there a reason the loader needs to be included with every js file inside `app` and it can't be included in just one of them, or a select few?

**Comment by dragonwocky** [comment]
@davidbailey00 yes - by including it in every js file, `module.export` functions for things like e.g. creating system menus or windows, and means things can be run in/at different processes/times/places (e.g. main and renderer, in the host window or in the notion webview).

`node_modules` is a good thought.

**Comment by dragonwocky** [comment]
a couple relevant updates:

i have added a `--path=</path/to/notion/resources/>` option, but the default is still the normal automatic platform detection.

so that modules can be updated/installed without having to update the whole enhancer, they'll be loaded from github and cached rather than distributed with the npm package (managed from the enhancer's menu).

i had originally planned for that cache to be in `~/.notion-enhancer`, but i did some testing to see if it would work to just add an option to install in "portable" mode and stick the cache somewhere in the notion source, and it wouldn't really. that would mean that if notion ever updates it would be deleted and users would not only have to reinstall the enhancer but also redownload and reconfigure all their modules.

it's not a massive issue, i can stick with the `~/.notion-enhancer` plan and just not use that cache during `notion-enhancer apply`, but it does mean that installations and pre-enhanced distributions can't come with any non-core mods pre-downloaded.
